vendor/pchart/pChart.class.php

```diff
@@ -2783,7 +2783,7 @@ function drawGraphAreaGradient($R,$G,$B,$Decay,$Target=TARGET_GRAPHAREA)
         {
          $R+=1;$G+=1;$B+=1;
          $C_Background = $this->AllocateColor($this->Picture,$R,$G,$B);
-         imagefilledrectangle($this->Picture,(int) round($X1),(int) round($Yi1),(int) round($X2),(int) round($Yi2),$C_Background);
+         imagefilledrectangle($this->Picture,(int) round($X1),(int) round($Yi1),(int) round((float) $X2),(int) round($Yi2),$C_Background);
```

Comment:
etc

Reply:
`         imagefilledrectangle($this->Picture,intval($X1),intval($Yi1),intval($X2),intval($Yi2),$C_Background);`

Else is just do the same thing with 2 cast + 1 function, and potentially broke with debug 1 and `$value === null` (or string)